homeassistant/components/lcn/binary_sensor.py

```diff
@@ -28,7 +36,11 @@ def create_lcn_binary_sensor_entity(hass, entity_config, config_entry):
     return LcnLockKeysSensor(entity_config, config_entry.entry_id, device_connection)
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    config_entry: ConfigEntry,
+    async_add_entities: Callable[[list[LcnEntity]], None],
```

Comment:
```suggestion
    async_add_entities: AddEntitiesCallback,
```

Same for other components